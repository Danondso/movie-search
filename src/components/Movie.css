.movie {
  height: auto;
  max-height: 400px;
  width: auto;
  max-width: 250px;
  padding: 1rem;
}

.movie-container {
  width: 100%;
}

.title {
  max-width: 20rem;
  padding: 1rem;
  font-size: 1.95rem;
}
