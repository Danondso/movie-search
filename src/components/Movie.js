import React from "react";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import { makeStyles } from "@material-ui/core/styles";

const DEFAULT_PLACEHOLDER_IMAGE =
  "https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_SX300.jpg";

const useStyles = makeStyles(theme => ({
  movie: {
    height: "auto",
    maxHeight: "400px",
    width: "auto",
    maxWidth: "250px",
    padding: "1rem"
  },
  title: {
    maxWidth: "20rem",
    padding: "1rem",
    fontSize: "1.95rem"
  }
}));

const Movie = ({ movie }) => {
  const classes = useStyles();
  const poster =
    movie.Poster === "N/A" ? DEFAULT_PLACEHOLDER_IMAGE : movie.Poster;
  return (
    <Card>
      <CardContent>
        <img alt={movie.Title} src={poster} className={classes.movie}></img>
        <div key={movie.Title} className={classes.title}>{movie.Title}</div>
      </CardContent>
    </Card>
  );
};

export default Movie;
