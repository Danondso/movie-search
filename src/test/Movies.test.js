import React from "react";
import Movie from "../components/Movie";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

const movieResult = {
  Title: "A fake movie",
  Year: "1983",
  imdbID: "imdbId",
  Type: "movie",
  Poster:
    "https://m.media-amazon.com/images/M/MV5BOWZlMjFiYzgtMTUzNC00Y2IzLTk1NTMtZmNhMTczNTk0ODk1XkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg"
};

const movieResultNoResult = {
  Title: "Star Wars: Episode XX - Return of the Franchise",
  Year: "1983",
  imdbID: "imdbId",
  Type: "movie",
  Poster: "N/A"
};

test("Movie renders with passed in data", () => {
  const component = shallow(<Movie movie={movieResult} />);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test("Movie without poster renders", () => {
  const component = shallow(<Movie movie={movieResultNoResult} />);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
