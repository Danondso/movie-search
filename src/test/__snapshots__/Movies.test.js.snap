// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders with passed in data 1`] = `
<img
  alt="A fake movie"
  className="movie"
  src="https://m.media-amazon.com/images/M/MV5BOWZlMjFiYzgtMTUzNC00Y2IzLTk1NTMtZmNhMTczNTk0ODk1XkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg"
/>
`;

exports[`Movie without poster renders 1`] = `
<img
  alt="Star Wars: Episode XX - Return of the Franchise"
  className="movie"
  src="https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_SX300.jpg"
/>
`;
